docs/assembly.rst

```diff
@@ -263,8 +263,9 @@ Here is an assembly snippet you can use for allocating memory that follows the p
     }
 
 The first 64 bytes of memory can be used as "scratch space" for short-term
-allocation. The 32 bytes after the free memory pointer (i.e., starting at ``0x60``)
-are meant to be zero permanently and is used as the initial value for
+allocation. The free memory pointer is itself 32 bytes. The following 32 bytes
+after the free memory pointer (i.e., starting at ``0x60``)
+are meant to be zero permanently, and they are used as the initial value for
 empty dynamic memory arrays.
```

Comment:
> he 32 bytes after the free memory pointer (i.e., starting at ``0x60``) are meant to be zero permanently and is used as the initial value forempty dynamic memory arrays.

I don't think this statement is true anymore.

Reply:
This is mentioned also in the Layout in Memory section. If it's no longer true, should this description be left in, or perhaps changed to say something like "this memory should be left zeroed out for legacy reasons."?

Reply:
> > he 32 bytes after the free memory pointer (i.e., starting at `0x60`) are meant to be zero permanently and is used as the initial value forempty dynamic memory arrays.
> 
> I don't think this statement is true anymore.

As a matter of fact, the statement still is true (in legacy codegen and also via-IR):
https://github.com/ethereum/solidity/blob/31b548577994397f7f849e8dd005d9d7aaa48c76/libsolidity/codegen/YulUtilFunctions.cpp#L4121